### PR TITLE
Upstream review of EuG - chapter 3

### PR DESCRIPTION
**Pull request by MichalMaler**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

File: modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc [hunk 1]
```diff
@@ -47,6 +47,12 @@ image::workspaces/workspace-devfile.png[]
 [id="defining-specific-container-images_{context}"]
 == Defining specific container images
 
+.Prerequisites
+
+* A running instance of {prod}. To install an instance of {prod}, see {link-installing-an-instance}.
+
+* An existing workspace defined on this instance of {prod}. See xref:creating-and-configuring-a-new-workspace.adoc[Creating and configuring a new {prod-short} {prod-ver} workspace].
```
Comment (boczkowska) [inline modules/end-user-guide/partials/proc_configuring-the-workspace-and-adding-tooling.adoc:54]:
```suggestion
* An existing workspace defined in this instance of {prod}. See xref:creating-and-configuring-a-new-workspace.adoc[Creating and configuring a new {prod-short} {prod-ver} workspace].
```

File: modules/end-user-guide/partials/ref_devfile-reference.adoc [hunk 2]
```diff
@@ -855,7 +857,7 @@ The following are two kinds of commands:
 
 Each {prod-short}-specific command features:
 
- * An `action` attribute that is a command to execute.
+ * An `actions` attribute that is a command to execute.
```
Comment (yhontyk) [inline modules/end-user-guide/partials/ref_devfile-reference.adoc]:
```suggestion
 * An `actions` attribute that specifies a command to execute.
```

File: modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc [hunk 3]
```diff
@@ -35,20 +35,7 @@ image::workspaces/che-config-projects-add-git.png[Add Git Project, link="../_ima
...
-----
-projects:
-  - name: che
-    source:
-      type: git
-      location: 'https://github.com/eclipse/che.git'
-----
-See the xref:making-a-workspace-portable-using-a-devfile.adoc#devfile-reference_{context}[].
-====
+For demonstration example, see xref:adding-a-git-project-into-a-workspace-using-a-devfile_{context}[Adding a .git project into a workspace using a devfile].
```
Comment (yhontyk) [inline modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc]: I'm not sure if it's the most modular way to add xrefs instead of specific examples. Could we maybe put that devfile in examples/ and reuse it in different procedures?

File: modules/end-user-guide/partials/proc_creating-a-new-workspace-from-the-dashboard.adoc [hunk 4]
```diff
@@ -17,22 +17,27 @@ To edit the devfile:
...
 
-. Scroll down to the *Devfile* section. In the *Devfile editor*, add necessary changes.
+. Scroll down to the *Devfile* section and edit the devfile using *Devfile editor*.
 +
 image::workspaces/minimal-devfile.png[Devfile editor with the minimal devfile, link="../_images/workspaces/minimal-devfile.png"]
 +
-[NOTE]
-.Example: add a project
-====
+For an example demonstration, see below.
```
Comment (yhontyk) [inline modules/end-user-guide/partials/proc_creating-a-new-workspace-from-the-dashboard.adoc]: I think "example demonstration" is a tautology. I suggest rewording to smth like "See an example below"

File: modules/end-user-guide/partials/proc_creating-a-new-workspace-from-the-dashboard.adoc [hunk 5]
```diff
@@ -17,22 +17,27 @@ To edit the devfile:
...
++
+[id="adding-a-git-project-into-a-workspace-using-a-devfile_{context}"]
+.Example - Adding a `.git` project into a workspace using a devfile
++
+In the following example, the project `{project-context}` acts as the example of a user's project. A user specifies this project using the `name` attribute of a devfile. The `location` attribute defines the source repository represented by an URL to a Git repository or ZIP archive.
```
Comment (yhontyk) [inline modules/end-user-guide/partials/proc_creating-a-new-workspace-from-the-dashboard.adoc]: "In the example... acts as an example". Suggest dropping one or rewording to smth like "In this section..."

File: modules/end-user-guide/partials/ref_devfile-reference.adoc (hunk 2 above)
Comment (MichalMaler) [reply modules/end-user-guide/partials/ref_devfile-reference.adoc]: Nice, will do

File: modules/end-user-guide/partials/proc_creating-a-new-workspace-from-the-dashboard.adoc (hunk 5 above)
Comment (MichalMaler) [reply modules/end-user-guide/partials/proc_creating-a-new-workspace-from-the-dashboard.adoc]: I changed it to  "instance"

File: modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc (hunk 3 above)
Comment (MichalMaler) [reply modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc]: What I did there was that I took away two bad examples, create one that is universal, place it in the first procedure, and xrefed it to another, since these two procedures are on the same page and really close to each other. I also wanted to remove the repetition, so I just pointed back + it is not some advanced stuff, so maybe it's not even needed it there, but since it was there originally, I at least used xref. let me know WDYT

Comment (yhontyk) [reply modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc]: I think we can just reuse that example. P2 is going to be displaying docs as separate procedures rather than the whole chapter, would be nice to start preparing for it :D

I'll pre-approve though. Everything else looks great.

**Review by yhontyk — APPROVED** [review]

File: modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc (hunk 3 above)
Comment (MichalMaler) [reply modules/end-user-guide/partials/proc_adding-projects-to-your-workspace.adoc]: Ok, I will patch it :) Thx for the review. Hope I will downstream this today. Need to have CRW ready up to section 3.6, where developer squat jumps in :)
